Orange/widgets/unsupervised/tests/test_owdistances.py

```diff
@@ -62,3 +64,23 @@ def test_mahalanobis_error(self):
                 self.send_signal("Data", data2)
                 self.assertEqual(self.widget.Error.mahalanobis_error.is_shown(), bad2)
                 self.assertEqual(self.get_output("Distances") is not None, out2)
+
+    def test_too_big_array(self):
+        """
+        Users sees an error message when calculating too large arrays and Orange
+        does not crash.
+        GH-2315
+        """
+        self.assertEqual(len(self.widget.Error.active), 0)
+        self.send_signal("Data", self.iris)
+
+        mock = Mock()
+        mock.side_effect = ValueError
+        self.widget.compute_distances(mock, self.iris)
```

Comment:
Use `side_effect` as a keyword argument for mock. I wouldn't even bother to have a `mock` variable here; I'd use `Mock(side_effect=ValueError)` as an argument.